src/delta.rs

```diff
@@ -413,7 +416,9 @@ fn handle_hunk_header_line(
     let (raw_code_fragment, line_numbers) = parse::parse_hunk_header(&line);
     // Emit the hunk header, with any requested decoration
     if config.hunk_header_style.is_raw {
-        writeln!(painter.writer)?;
+        if config.hunk_header_style.decoration_style != DecorationStyle::NoDecoration {
+            writeln!(painter.writer)?;
+        }
```

Comment:
@yoichi do you recall the logic here -- why do we want a newline only if `hunk_header_style.decoration_style != DecorationStyle::NoDecoration`? I'm sorry, I should have asked you to add a comment at the time. Given that `hunk-header-style=omit` is failing (#274), I'm thinking we need to do more work to characterize the conditions under which the newline and other alterations are permitted. If you have any thoughts then great, but if not don't worry!

Reply:
Thanks @yoichi! I wonder whether I should get rid of that extra newline character completely, at least in order to get the code into an easy-to-understand state. And then consider re-introducing it. I think I added it in the big v0.2 update because I thought that the extra space made the diff easier to read.